# lowlevel controller
llc_config_file: src/unitree_lowlevel/config/g1.yaml

# Mujoco deploy.py: model.opt.timestep = 0.005, decimation=4 -> policy_dt=0.02
ll_dt: 0.005

policy:
  backend: ort
  model_path: src/legged_rl_deploy/policies/g1-velocity-unitree_rl_mjlab/policy.onnx
  input_dim: 96
  output_dim: 29
  policy_dt: 0.02

  joint_ids_map: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28]
  stiffness: [40.2, 99.1, 40.2, 99.1, 28.5, 28.5, 40.2, 99.1, 40.2, 99.1, 28.5, 28.5, 40.2, 28.5, 28.5,
              14.3, 14.3, 14.3, 14.3, 14.3, 16.8, 16.8, 14.3, 14.3, 14.3, 14.3, 14.3, 16.8, 16.8]
  damping:   [2.6, 6.3, 2.6, 6.3, 1.8, 1.8, 2.6, 6.3, 2.6, 6.3, 1.8, 1.8, 2.6, 1.8, 1.8,
              0.9, 0.9, 0.9, 0.9, 0.9, 1.1, 1.1, 0.9, 0.9, 0.9, 0.9, 0.9, 1.1, 1.1]

  commands:
    base_velocity:
      process_order: [clip]
      min: [-0.5, -0.5, -1.0]
      max: [1.0, 0.5, 1.0]

  actions:
    JointPositionAction:
      process_order: [scale, offset]
      scale: [0.55, 0.35, 0.55, 0.35, 0.44, 0.44, 0.55, 0.35, 0.55, 0.35, 0.44, 0.44, 0.35, 0.44, 0.44,
              0.44, 0.44, 0.44, 0.44, 0.44, 0.07, 0.07, 0.44, 0.44, 0.44, 0.44, 0.44, 0.07, 0.07]
      offset: [
        -0.1,0,0,0.3,-0.2,0, 
        -0.1,0,0,0.3,-0.2,0,  
        0,0,0,  
        0.35,0.18,0,0.87,0,0,0, 
        0.35,-0.18,0,0.87,0,0,0
      ]
 
  observations:
    stack:
      length: 1
      order: newest_first
    terms:
      # 3 + 3 + 3 + 29 + 29 + 29 = 96
      base_ang_vel_B:
      projected_gravity:
      velocity_commands:
      joint_pos:
        process_order: [offset]
        offset: [
          0.1, 0, 0, -0.3, 0.2, 0, 
          0.1, 0, 0, -0.3, 0.2, 0, 
          0, 0, 0, 
          -0.35, -0.18, 0, -0.87, 0, 0, 0, 
          -0.35, 0.18, 0, -0.87, 0, 0, 0
        ]
      joint_vel:
      last_action:

